therapy_app/client.py

```diff
@@ -0,0 +1,175 @@
+from django.db.models import Q
+from rest_framework.views import APIView, Response, status
+from . import models
+from .authentication_classes import CsrfExemptSessionAuthentication
+from .permissions import *
+from . import serializers
+
+
+# Create your views here.
+class SearchTherapistView(APIView):
+    """
+    This APIView takes GET request only to show available therapists to clients.
+    This only shows those therapists that are not actively mapped to the requesting client.
+    """
+    authentication_classes = (CsrfExemptSessionAuthentication,)
+    permission_classes = [IsClient]
+
+    def get(self, request, format=None):
+        """
+        This method will return the list of therapists that are not actively mapped to the requesting client.
+        """
+        therapists = models.User.objects.filter(is_deleted=False, type='Therapist').exclude(
+            Q(mapping_therapist__client=request.user) & Q(mapping_therapist__is_deleted=False))
+        serialized = serializers.UserSerializer(therapists, many=True)
+        return Response(serialized.data, status=status.HTTP_200_OK)
+
+
+class MapTherapistView(APIView):
+    """
+    This APIView takes GET request only to map available therapists to clients.
+    This only maps those therapists that are not actively mapped to the requesting client.
+    takes DELETE request to remove mapping of a therapist and requesting client.
+    """
+    authentication_classes = (CsrfExemptSessionAuthentication,)
+    permission_classes = [IsClient]
+
+    def get(self, request, id1, format=None):
+        """
+        This method will map a therapist that is not actively mapped to the requesting client.
+        """
+        if models.User.objects.filter(is_deleted=False, type='Therapist', pk=id1).exists():
+            try:
+                therapist = models.User.objects.filter(is_deleted=False, type='Therapist').exclude(
+                        Q(mapping_therapist__client=request.user) & Q(mapping_therapist__is_deleted=False)).get(pk=id1)
```

Comment:
Since the query to get relevant therapists for a client is showing up again and again - it would make sense to make this a part of the User model object itself right?